### PR TITLE
Consider .hsc files to be Haskell modules

### PR DESCRIPTION
**Pull request by mitchellwrosen**

### CODE REVIEW

File: test/Hpack/UtilSpec.hs [hunk 1]
```diff
@@ -38,7 +38,9 @@ spec = do
 
   describe "toModule" $ do
     it "maps paths to module names" $ do
-      toModule ["Foo", "Bar", "Baz.hs"] `shouldBe` Just "Foo.Bar.Baz"
+      toModule ["Foo", "Bar", "Baz.hs"]  `shouldBe` Just "Foo.Bar.Baz"
+      toModule ["Foo", "Bar", "Baz.lhs"] `shouldBe` Just "Foo.Bar.Baz"
+      toModule ["Foo", "Bar", "Baz.hsc"] `shouldBe` Just "Foo.Bar.Baz"
```
Comment (sol) [inline test/Hpack/UtilSpec.hs]: I think it's a good thing to unit tests this, but this should be 3 test cases, one expectation per test.

File: test/Hpack/ConfigSpec.hs [hunk 2]
```diff
@@ -815,8 +815,9 @@ spec = do
           touch "src/Foo.hs"
           touch "src/Bar.hs"
           touch "src/Baz.lhs"
+          touch "src/Qux.hsc"
           )
-          (map (executableOtherModules . sectionData) . packageExecutables >>> (`shouldBe` [["Bar", "Baz", "Foo"]]))
+          (map (executableOtherModules . sectionData) . packageExecutables >>> (`shouldBe` [["Bar", "Baz", "Foo", "Qux"]]))
```
Comment (sol) [inline test/Hpack/ConfigSpec.hs]: At the point we have unit tests for this, we don't need to test all three cases up here anymore.  just doing it for e.g. `.hs` is fine.

**Comment by sol** [comment]
Other than my comments it LGTM.

**Comment by mitchellwrosen** [comment]
Done!

File: test/Hpack/UtilSpec.hs [hunk 3]
```diff
@@ -36,9 +36,17 @@ spec = do
     it "accepts qualified identifier" $ do
       parseMain "Foo.bar" `shouldBe` ("Foo.hs", ["-main-is Foo.bar"])
 
-  describe "toModule" $ do
```
Comment (sol) [inline test/Hpack/UtilSpec.hs]: let's keep his line

File: test/Hpack/UtilSpec.hs [hunk 4]
```diff
@@ -36,9 +36,17 @@ spec = do
...
-    it "maps paths to module names" $ do
-      toModule ["Foo", "Bar", "Baz.hs"] `shouldBe` Just "Foo.Bar.Baz"
+  describe ".hs toModule" $ do
```
Comment (sol) [inline test/Hpack/UtilSpec.hs]: drop this line

File: test/Hpack/UtilSpec.hs [hunk 5]
```diff
@@ -36,9 +36,17 @@ spec = do
...
+    it "maps .hs paths to module names" $ do
+      toModule ["Foo", "Bar", "Baz.hs"]  `shouldBe` Just "Foo.Bar.Baz"
+
+  describe ".lhs toModule" $ do
```
Comment (sol) [inline test/Hpack/UtilSpec.hs]: drop this line

File: test/Hpack/UtilSpec.hs [hunk 6]
```diff
@@ -36,9 +36,17 @@ spec = do
...
+    it "maps .lhs paths to module names" $ do
+      toModule ["Foo", "Bar", "Baz.lhs"] `shouldBe` Just "Foo.Bar.Baz"
+
+  describe ".hsc toModule" $ do
```
Comment (sol) [inline test/Hpack/UtilSpec.hs]: drop this line

**Comment by sol** [comment]
Specifically, I follow the convention to have one `describe` per function.  If I need more structure, I use `context`, but I think we don't necessarily need it here.

**Comment by mitchellwrosen** [comment]
Ah, I misunderstood. Moment...

**Comment by mitchellwrosen** [comment]
Ok, done

**Pull request merged**